packages/block-library/src/button/edit.js

```diff
@@ -115,34 +115,29 @@ function useEnter( props ) {
 }
 
 function WidthPanel( { selectedWidth, setAttributes } ) {
-	function handleChange( newWidth ) {
-		// Check if we are toggling the width off
-		const width = selectedWidth === newWidth ? undefined : newWidth;
-
-		// Update attributes.
-		setAttributes( { width } );
-	}
-
 	return (
 		<PanelBody title={ __( 'Settings' ) }>
-			<ButtonGroup aria-label={ __( 'Button width' ) }>
+			<ToggleGroupControl
+				label={ __( 'Button width' ) }
+				value={ selectedWidth }
+				hideLabelFromVision
+				onChange={ ( newWidth ) =>
+					setAttributes( { width: newWidth } )
+				}
+				isDeselectable
```

Comment:
@t-hamano suggested in https://github.com/WordPress/gutenberg/issues/52223#issuecomment-2274853258 that we add an "Auto" or "None" option for the reset. I think that might be better than the `isDeselectable` variant in this case, because our options items here are not icon buttons.

Reply:
Should we address this in a follow-up as @t-hamano [suggested](https://github.com/WordPress/gutenberg/pull/65346#issuecomment-2353320789)?

There's another discussion for using "None" options, not sure if it affects this case. See #64906.

Reply:
Mmm, good point about the "None" discussion.

The thing is, "Text options + `isDeselectable`" is [not a valid pattern](https://github.com/WordPress/gutenberg/pull/65346#discussion_r1762374255) for this component 😕 For the sake of this PR, could we do a "None" option, or maybe add a separate Clear or Reset button?

Reply:
To be honest, the reason I suggested adding Auto/None options [here](https://github.com/WordPress/gutenberg/issues/52223#issuecomment-2274853258) was because I wasn't aware of the `isDeselectable` prop 😅

I also learned from this discussion that "Text option + `isDeselectable`" is not a valid pattern.

When an explicit percentage width is not set on a button, the button's width grows and shrinks depending on the text inside, so "Auto" feels more appropriate than "None".

Reply:
Sounds good! I guess we should go with a adding a "None"  option?

Reply:
"Auto" should work fine, but as mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/65340#issuecomment-2360428325), it may cause unintended layouts when localized.

![button](https://github.com/user-attachments/assets/15a849c4-7e27-40fd-9ad7-2f50feac0a41)

If there was a custom width (UnitControl) UI, we could set the value to empty (`undefined`) via that, but currently there isn't.

Given this, would it be a good idea to add a reset button?

Reply:
As explained in https://github.com/WordPress/gutenberg/pull/65340#issuecomment-2360651360, if we currently added a "Reset" button, we'd still face the issue that `ToggleGroupControl` selects a value when it receives focus.

So, with the current behaviour of the component, an `Auto` option is the only viable option IMO. We could otherwise pause the work here until `ToggleGroupControl` gets a new design and behaviour.